lucien-ios/APIClient/Request.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  Request.swift
+//
+//  Created by Mark Daigneault on 5/3/16.
+//  Copyright © 2016 Intrepid Pursuits. All rights reserved.
+//
+
+import Foundation
+
+public enum HTTPMethod: String {
+    case GET
+    case POST
+    case PATCH
+    case PUT
+    case DELETE
+}
+
+public protocol Request {
+    static var baseURL: String { get }
+    static var acceptHeader: String? { get }
+    static var authorizationHeader: String? { get }
+
+    var method: HTTPMethod { get }
+    var path: String { get }
+    var authenticated: Bool { get }
+    var queryParameters: [String: Any]? { get }
+    var bodyParameters: [String: Any]? { get }
+    var contentType: String { get }
+//    var accessCredentials: AccessCredentials? { get }
+}
+
+public extension Request {
+
+    var urlRequest: URLRequest {
+        let baseURL = Foundation.URL(string: Self.baseURL)!
+        let url = Foundation.URL(string: path, relativeTo: baseURL) ?? baseURL
+
+        var request = URLRequest(url: url)
+        request.httpMethod = method.rawValue
+
+        request.setValue(Self.acceptHeader, forHTTPHeaderField: "Accept")
+        request.setValue(Self.authorizationHeader, forHTTPHeaderField: "Authorization")
+        request.setValue(contentType, forHTTPHeaderField: "Content-Type")
+
+        if authenticated {
+//            accessCredentials?.authorize(&request)
```

Comment:
Commented-out code shouldn't be in the PR. If it's not being used, it can just be deleted